tests/integration/components/self-clear-cache-test.js

```diff
@@ -0,0 +1,127 @@
+  setupRenderingTest(hooks);
+
+  skip('it shows last time user cleared the cache', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    await render(hbs`<SelfClearCache @lastClearedOn={{this.lastTime}} />`);
+
+    // Assert
+    assert.dom('[data-test-last-time]').hasText(HUMAN_TIME);
+  });
+
+  skip('it triggers cache clear on button click', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    const CACHE_CLEAR_CLICKED = 'CACHE_CLEAR_CLICKED';
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    this.set('onCacheClear', function () {
+      assert.step(CACHE_CLEAR_CLICKED);
+    });
+
+    await render(hbs`
+      <SelfClearCache
+        @lastClearedOn={{this.lastTime}}
+        @onCacheClear={{this.onCacheClear}}
+      />
+    `);
+
+    // Act
+    const btn = find('[data-test-btn-clear-cache]');
+    await click(btn);
+
+    // Assert
+    assert.verifySteps([CACHE_CLEAR_CLICKED]);
+  });
+
+  skip('it disables button after a click', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    const CACHE_CLEAR_CLICKED = 'CACHE_CLEAR_CLICKED';
+    this.set('lastTime', EPOCH_TIMESTAMP);
+
+    this.set('onCacheClear', function () {
+      assert.step(CACHE_CLEAR_CLICKED);
+    });
+
+    await render(hbs`
+      <SelfClearCache
+        @lastClearedOn={{this.lastTime}}
+        @onCacheClear={{this.onCacheClear}}
+      />
+    `);
+
+    // Act
+    const btn = find('[data-test-btn-clear-cache]');
+    await click(btn);
+    await click(btn); // Another click -- This should not fire
+
+    // Assert
+    assert.verifySteps([CACHE_CLEAR_CLICKED]);
+  });
+
+  skip('it shows the number of times cache has already been cleared', async function (assert) {
+    assert.expect(1);
+
+    // Assemble
+    this.set('lastTime', EPOCH_TIMESTAMP);
+    this.set('totalTimes', 2);
+    this.set('allowedLimit', 3);
+
+    await render(hbs`
+      <SelfClearCache
+        @time={{this.lastTime}} 
+        @countCleared={{this.totalTimes}}
+        @allowedLimit={{this.maxallowedLimit}}
+      />
+    `);
+
+    // Assert
+    assert.dom('[data-test-pending-requests]').hasText('2 / 3');
+  });
+
+  skip('it disables the button if already reached allowed limit', async function (assert) {
+    assert.expect(2);
+
+    // Assemble
+    const CACHE_CLEAR_CLICKED = 'CACHE_CLEAR_CLICKED';
+
+    this.set('lastTime', EPOCH_TIMESTAMP);
+    this.set('totalTimes', 3);
+    this.set('allowedLimit', 3);
+
+    this.set('onCacheClear', function () {
+      assert.step(CACHE_CLEAR_CLICKED);
+    });
+
+    await render(hbs`
+      <SelfClearCache
+        @time={{this.lastTime}} 
+        @countCleared={{this.totalTimes}}
+        @allowedLimit={{this.maxallowedLimit}}
+        @onCacheClear={{this.onCacheClear}}
+      />
+    `);
+
+    // Act
+    const btn = find('[data-test-btn-clear-cache]');
+    await click(btn);
+
+    // Assert
+    assert.verifySteps([]);
```

Comment:
As per my understanding, on click of button for third consecutive time, button needs to be disabled as it has reached max Allowed limit, but how are we doing the above action,
Please correct me if my thought process is wrong

Reply:
This checks if any click steps occured

Reply:
The component is being told what is the max times the cache can be cleared, and how many times the users has already cleared it.
The expectation is that the rendered button should not be firing click actions (how does not matter for the tests). 

The button is given a function for onClick event, but the test asserts that even if the function for onClick exists on the button, if the user has exhausted their quote for clearing the cache, the function on the button should not trigger (even though it exists)